.github/workflows/publish.yml

```diff
@@ -9,6 +9,7 @@ jobs:
   push_to_registry:
     name: Push Docker image to Docker Hub
     runs-on: ubuntu-latest
+    environment: releases
```

Comment:
not sure - will it fallback to non-env secrets if they not found in environment? like DOCKER_USERNAME

can't CRATESIO_TOKEN be non-env?

Reply:
> not sure - will it fallback to non-env secrets if they not found in environment? like DOCKER_USERNAME

it should

> can't CRATESIO_TOKEN be non-env?

This was requested by security, and it's totally justified IMO. "Releases" environment is limited to `v*` tags only, same as this workflow.